docs/DeveloperGuide.md

```diff
@@ -2,37 +2,243 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* This application is built in partial fulfillment of the requirements of CS2113.
+* Knowledge and ideas are adapted from the CS2113 textbook, as well as SE-EDU textbook.
+
+---
+
+## Setting up, getting started
+
+Refer to the User guide at https://ay2324s2-cs2113-w14-2.github.io/tp/UserGuide.html.
+
+---
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" width="350" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the application.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+**`RecipeIO`**  is in charge of the app launch and shut down.
+* At app launch, it initializes the other components in the correct sequence, and connects them up with each other.
+* At shut down, it shuts down the other components and invokes cleanup methods where necessary.
+
+The bulk of the app's work is done by the following five components:
+
+* `UI` The UI of the App.
+* `RecipeList`: The command executor and holder of data of the app.
+* `InputParser`: Extracts information from the user input into the command line.
+* `CommandValidator`: The command validator.
+* `Storage`: Reads data from, and writes data to, the hard disk.
+
+the `commands`package represents a collection of commands used by **`RecipeList`**
```

Comment:
Something minor but may want to edit this to caps the first letter and end the sentence with a full stop.